### PR TITLE
mam/v2: wots - calc_pks

### PR DESCRIPTION
**Pull request by tsvisabo**
- fix `set_trit_at` on uninitialized dest arrays
- wots_calc_pks: trits_t -> trit_array_t
-  add `trit_array_insert_from_pos` that takes poistions in both src/dest arrays

### CODE REVIEW

File: mam/v2/wots.c [hunk 1]
```diff
@@ -14,23 +14,31 @@
...
 
-static void wots_calc_pks(isponge *const sponge, trits_t sk_pks, trits_t pk) {
+static void wots_calc_pks(isponge *const sponge, trit_array_p sk_pks,
+                          trit_array_p pk) {
   size_t i, j;
-  trits_t sk_part, pks = sk_pks;
 
-  MAM2_ASSERT(trits_size(sk_pks) == MAM2_WOTS_SK_SIZE);
-  MAM2_ASSERT(trits_size(pk) == MAM2_WOTS_PK_SIZE);
+  TRIT_ARRAY_DECLARE(sk_part_trits_array, 0);
```
Comment (thibault-martinez) [inline mam/v2/wots.c]: Why don't you just `TRIT_ARRAY_DECLARE(sk_part_trits_array, MAM2_WOTS_SK_PART_SIZE);` instead if size 0 then setting an array ?

File: mam/v2/wots.c [hunk 2]
```diff
@@ -141,12 +149,21 @@ void wots_gen_sk3(wots_t *const wots, prng_t *const prng, trits_t const nonce1,
 }
 
 void wots_calc_pk(wots_t *const wots, trits_t pk) {
-  MAM2_TRITS_DEF(sk_pks, MAM2_WOTS_SK_SIZE);
-  // TODO Remove when wots_calc_pks takes flex_trit_t *
-  flex_trits_to_trits(sk_pks.p + sk_pks.d, MAM2_WOTS_SK_SIZE, wots->sk,
-                      MAM2_WOTS_SK_SIZE, MAM2_WOTS_SK_SIZE);
-  wots_calc_pks(wots->sponge, sk_pks, pk);
-  trits_set_zero(sk_pks);
+  TRIT_ARRAY_DECLARE(sk_trits_array, 0);
```
Comment (thibault-martinez) [inline mam/v2/wots.c]: Same question

File: mam/v2/wots.c [hunk 3]
```diff
@@ -141,12 +149,21 @@ void wots_gen_sk3(wots_t *const wots, prng_t *const prng, trits_t const nonce1,
...
+  flex_trit_t sk_flex_trits[trit_array_bytes_for_trits(MAM2_WOTS_SK_SIZE)];
+  memcpy(sk_flex_trits, wots->sk, MAM2_WOTS_SK_FLEX_SIZE);
+  trit_array_set_trits(&sk_trits_array, sk_flex_trits, MAM2_WOTS_SK_SIZE);
+
+  TRIT_ARRAY_DECLARE(pk_trits_array, 0);
```
Comment (thibault-martinez) [inline mam/v2/wots.c]: Same question

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: mam/v2/wots.c (hunk 1 above)
Comment (tsvisabo) [reply mam/v2/wots.c]: oh, that's stupid

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**